### ISSUE TITLE
Model.fit in keras style python API doesn't support both x and y being RDDs

### ISSUE THREAD

**Issue by fanzhuyifan** [intel-analytics/analytics-zoo#2918]
<https://github.com/intel-analytics/analytics-zoo/blob/35fd8d19754296e6e8a9dcdd689c15e88a2e1dab/pyzoo/zoo/pipeline/api/keras/engine/topology.py#L217-L233>

Here, when calling fit, if both `x` and `y` are RDDs, an `TypeError("Unsupported training data type: %s" % type(x))` would be thrown.

Also, it would be nice if the documentation specifies the format of the single RDD to be passed, i.e., how the labels and features should be specified, in the case where `y=None`.

**Comment by hkvision** [comment]
Hi @fanzhuyifan 

We have written in the documentation that x needs to be RDD of Sample in the case y is None. Thanks for pointing out and I will make the documentation clearer.

**Comment by fanzhuyifan** [comment]
现在Model.fit不支持x和y同时是RDD，希望能有支持！

如果x和y都是RDD的话，会引发`TypeError("Unsupported training data type: %s" % type(x))`

Right now Model.fit doesn't support the case where both x and y are RDDs.
If they are both RDDs, a TypeError("Unsupported training data type: %s" % type(x))` will be thrown.

**Comment by hkvision** [comment]
What are you expected to input for x, y both RDDs? They are both RDD of ndarray or RDD of list of ndarrays?

**Comment by fanzhuyifan** [comment]
Sorry I just realized that the `TypeError` I was getting was in an older version of analytics-zoo. The current version doesn't raise this error when I call `model.fit(x=rdd1, y=rdd2)`.

However, `model.fit` still doesn't work when both x and y are RDDs of list of ndarrays. **If you look at the code I linked to, you will see that in the case where x is an RDD and y is also an RDD, they do not get merged together in `training_data`. Instead, only x is used.** Right now I am using `model.fit` with both x and y being RDDs, either RDDs of list of ndarrays, or RDDs of list of Rows. I could manually combine x and y by 
```
from zoo.pipeline.api.keras.utils import Sample
xRdd = xRdd.map(lambda element: np.array(element))
yRdd = yRdd.map(lambda element: np.array(element))
train_rdd = xRdd.zip(yRdd).map(lambda item: Sample.from_ndarray(item[0], item[1]))
```
But it seems rather stupid that `model.fit` can't handle this case automatically.

**Comment by hkvision** [comment]
Yeah, in our implementation, we only expect RDD of Sample as input. I will raise a PR to support for both x and y being RDDs. But I think currently we target at supporting RDD of ndarrays. Why do you have RDD of Rows? In this case, why don't use use just Spark DataFrame?

**Comment by fanzhuyifan** [comment]
Right now I am having RDD of Rows because the documentation says only np.ndarray and RDDs are supported. So I am just calling ".rdd" on the result from a spark.sql selection, which gives me RDD of Rows. Of course I would want to use `Model.fit` directly on a Spark DataFrame, but that doesn't seem supported right now.

**Comment by hkvision** [comment]
Hi @fanzhuyifan The Keras-sytle fit API is designed for Spark RDDs. For DataFrame, we have support for NNEstimator here https://github.com/intel-analytics/analytics-zoo/blob/master/pyzoo/zoo/pipeline/nnframes/nn_classifier.py#L135 and you may take a look.
Also, I have raised a PR to address your issue. You may take a look. Thank you so much!

**Comment by fanzhuyifan** [comment]
Thank you so much for your prompt reply and quick fixes! : )

**Comment by hkvision** [comment]
> es! : )

Thank you for using Analytics Zoo and give suggestions!

**Comment by helenlly** [comment]
@hkvision ,can we go ahead to close this issue?

**Comment by hkvision** [comment]
The PR #2920 is still under review.
